tests/test_addons.py

```diff
@@ -837,6 +838,28 @@ def test_delete_action_for_folder_deletes_subfolders_and_creates_trashed_file_no
         assert_false(StoredFileNode.load(subfolder._id))
 
 
+    def test_archived_from_url_with_id(self):
+        file_node = self.get_test_file()
+        file_node.archived_from_id = '12345'
+        registered_node = self.project.register_node(
+            schema=get_default_metaschema(),
+            auth=Auth(self.user),
+            data=None,
+        )
+        archived_from_url = views.get_archived_from_url(registered_node, file_node)
+        assert_true(archived_from_url)
+
+    def test_archived_from_url_without_id(self):
+        file_node = self.get_test_file()
+        registered_node = self.project.register_node(
+            schema=get_default_metaschema(),
+            auth=Auth(self.user),
+            data=None,
+        )
+        archived_from_url = views.get_archived_from_url(registered_node, file_node)
+        assert_false(archived_from_url)
+
```

Comment:
Can you also add a test to check that if a file node is in `TrashedFileNode`, it's archived_from_url is None?